v3/src/models/data/data-set.ts

```diff
@@ -180,6 +180,23 @@ export const DataSet = types.model("DataSet", {
   }
 })
 .views(self => ({
```

Comment:
I like how these two new views simplify things by removing need to maintain these maps. 👍🏻

Reply:
I was originally maintaining all of the maps manually out of concern for performance, but adding/removing attributes doesn't happen often enough and the numbers of attributes aren't great enough to warrant the concern. For now, I'm leaving the case IDs map maintained manually because rebuilding it from scratch on every change could be expensive with a large number of cases, whereas we can handle the most common case -- adding a new case to the end -- much more efficiently when responding manually. Will probably have to address that issue down the road as well.